src/mono/netcore/System.Private.CoreLib/src/System/Reflection/Emit/TypeBuilder.Mono.cs

```diff
@@ -1819,6 +1821,8 @@ public GenericTypeParameterBuilder[] DefineGenericParameters(params string[] nam
             return generic_params;
         }
 
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2006:UnrecognizedReflectionPattern",
+            Justification = "Type.GetConstructor(ConstructorInfo) is not actually unsafe")]
```

Comment:
Should we fix linker to not mark it as problematic then?

Reply:
This overload of GetConstructor is a Monoism that doesn't exist elsewhere.

The question is whether linker should check whether the signatures match the expectations based on the public surface area.

I don't know if I really want us to spend time checking all the signatures...

Reply:
OK - I'm fine with that - just add a comment why linker doesn't handle this "correctly"